master/internal/checkpoints/postgres_checkpoints_intg_test.go

```diff
@@ -16,25 +18,245 @@ import (
+	err := db.AddCheckpointMetadata(ctx, &checkpoint, tr.ID)
+	require.NoError(t, err)
+
+	// Insert a model.
+	now := time.Now()
+	mdl := model.Model{
+		Name:            uuid.NewString(),
+		Description:     "some important model",
+		CreationTime:    now,
+		LastUpdatedTime: now,
+		Labels:          []string{"some other label"},
+		Username:        user.Username,
+		WorkspaceID:     1,
+	}
+	mdlNotes := "some notes1"
+	var pmdl modelv1.Model
+	err = db.SingleDB().QueryProto(
+		"insert_model", &pmdl, mdl.Name, mdl.Description, emptyMetadata,
+		strings.Join(mdl.Labels, ","), mdlNotes, user.ID, mdl.WorkspaceID,
+	)
+	require.NoError(t, err)
+
+	var retCkpt1 checkpointv1.Checkpoint
+	err = db.SingleDB().QueryProto("get_checkpoint", &retCkpt1, checkpoint.UUID)
+	require.NoError(t, err)
+
+	addmv := modelv1.ModelVersion{
+		Model:      &pmdl,
+		Checkpoint: &retCkpt1,
+		Name:       "checkpoint 1",
+		Comment:    "empty",
+	}
+	var mv modelv1.ModelVersion
+	err = db.SingleDB().QueryProto(
+		"insert_model_version", &mv, pmdl.Id, retCkpt1.Uuid, addmv.Name, addmv.Comment,
+		emptyMetadata, strings.Join(addmv.Labels, ","), addmv.Notes, user.ID,
+	)
+	require.NoError(t, err)
+
+	expmodelIDsCheckpoint := []int32{pmdl.Id}
+	modelIDsCheckpoint, err := GetModelIDsAssociatedWithCheckpoint(context.TODO(), checkpoint.UUID)
+	require.NoError(t, err)
+	require.Equal(t, expmodelIDsCheckpoint, modelIDsCheckpoint)
+}
+
+func TestGetRegisteredCheckpoints(t *testing.T) {
+	ctx := context.Background()
+
+	user := db.RequireMockUser(t, db.SingleDB())
+
+	exp := db.RequireMockExperiment(t, db.SingleDB(), user)
+	tr, task := db.RequireMockTrial(t, db.SingleDB(), exp)
+	allocation := db.RequireMockAllocation(t, db.SingleDB(), task.TaskID)
+
+	ckpt1 := uuid.New()
+	checkpoint1 := db.MockModelCheckpoint(ckpt1, allocation)
+	err := db.AddCheckpointMetadata(ctx, &checkpoint1, tr.ID)
+	require.NoError(t, err)
+
+	ckpt2 := uuid.New()
+	checkpoint2 := db.MockModelCheckpoint(ckpt2, allocation)
+	err = db.AddCheckpointMetadata(ctx, &checkpoint2, tr.ID)
+	require.NoError(t, err)
+
+	ckpt3 := uuid.New()
+	checkpoint3 := db.MockModelCheckpoint(ckpt3, allocation)
+	err = db.AddCheckpointMetadata(ctx, &checkpoint3, tr.ID)
+	require.NoError(t, err)
+
+	// Insert a model.
+	now := time.Now()
+	mdl := model.Model{
+		Name:            uuid.NewString(),
+		Description:     "some important model",
+		CreationTime:    now,
+		LastUpdatedTime: now,
+		Labels:          []string{"some other label"},
+		Username:        user.Username,
+		WorkspaceID:     1,
+	}
+	mdlNotes := "some notes2"
+	var pmdl modelv1.Model
+	err = db.SingleDB().QueryProto(
+		"insert_model", &pmdl, mdl.Name, mdl.Description, emptyMetadata,
+		strings.Join(mdl.Labels, ","), mdlNotes, user.ID, mdl.WorkspaceID,
+	)
+	require.NoError(t, err)
+
+	var retCkpt1 checkpointv1.Checkpoint
+	err = db.SingleDB().QueryProto("get_checkpoint", &retCkpt1, checkpoint1.UUID)
+	require.NoError(t, err)
+
+	addmv1 := modelv1.ModelVersion{
```

Comment:
style nit (none-blocking): staying away from using verbs as variable names can improve readability especially when the variable type is implicitly defined; by no means necessary to change here

Reply:
done